core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -734,14 +742,28 @@ public void addColumn(Session session, TableHandle tableHandle, ColumnMetadata c
         CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
         ConnectorMetadata metadata = getMetadataForWrite(session, catalogHandle);
         metadata.addColumn(session.toConnectorSession(catalogHandle), tableHandle.getConnectorHandle(), column);
+
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogHandle);
+        if (catalogMetadata.getSecurityManagement() != CONNECTOR) {
+            TableMetadata tableMetadata = getTableMetadata(session, tableHandle);
+            CatalogSchemaTableName sourceTableName = new CatalogSchemaTableName(catalogHandle.getCatalogName(), tableMetadata.getTable());
+            systemSecurityMetadata.columnCreated(session, sourceTableName, column.getName());
+        }
     }
 
     @Override
     public void dropColumn(Session session, TableHandle tableHandle, ColumnHandle column)
     {
         CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
         ConnectorMetadata metadata = getMetadataForWrite(session, catalogHandle);
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogHandle);
         metadata.dropColumn(session.toConnectorSession(catalogHandle), tableHandle.getConnectorHandle(), column);
+        if (catalogMetadata.getSecurityManagement() != CONNECTOR) {
+            String columnName = getColumnMetadata(session, tableHandle, column).getName();
+            TableMetadata tableMetadata = getTableMetadata(session, tableHandle);
+            CatalogSchemaTableName sourceTableName = new CatalogSchemaTableName(catalogHandle.getCatalogName(), tableMetadata.getTable());
+            systemSecurityMetadata.columnDropped(session, sourceTableName, columnName);
```

Comment:
this may be problematic from transaction perspective. 
what if we dispatch the `systemSecurityMetadata.columnDropped` event but the transaction doesn't get committed eventually and we end up still having the column?

credits to @losipiuk for thinking about that

Reply:
a bit related, but vice versa, a failure here will cause Trino transaction failure even if the connector applied the changes immediately (many do). This led to https://github.com/trinodb/trino/issues/12535, see https://github.com/trinodb/trino/issues/12535#issuecomment-1417684880 for explanation

Reply:
Notice that problem also exists for other `systemSecurityMetadata` notification methods too, like `alter table rename`